src/Components/CreateImageWizard/formComponents/Packages.js

```diff
@@ -156,60 +163,99 @@ const Packages = ({ defaultArch, ...props }) => {
-            // set the steps field to the current chosen packages list
-            change(input.name, removePackagesDisplayFields(destinationPackages));
-        } else {
-            setPackagesChosen(updatedSourcePackages);
-            sortPackages(packagesAvailable);
-            // set the steps field to the current chosen packages list
-            change(input.name, removePackagesDisplayFields(updatedSourcePackages));
-        }
+        const updatedPackagesChosen = [ ...newPackagesChosen, ...packagesChosen ];
+
+        updateState(updatedPackagesAvailable, updatedPackagesChosen);
     };
 
-    // move all packages
-    const moveAll = (fromAvailable) => {
-        let chosenPackages = [];
-        if (fromAvailable) {
-            chosenPackages = [ ...packagesAvailable.filter(pack => !pack.isHidden), ...packagesChosen ];
-            setPackagesChosen(chosenPackages);
-            sortPackages([ ...packagesAvailable.filter(pack => pack.isHidden) ]);
-        } else {
-            chosenPackages = [ ...packagesChosen.filter(pack => pack.isHidden) ];
-            sortPackages([ ...packagesChosen.filter(pack => !pack.isHidden), ...packagesAvailable ]);
-            setPackagesChosen(chosenPackages);
-        }
+    const moveSelectedToAvailable = () => {
+        const newPackagesAvailable = [];
 
-        // set the steps field to the current chosen packages list
-        change(input.name, removePackagesDisplayFields(chosenPackages));
+        const updatedPackagesChosen = packagesChosen.filter((pack) => {
+            if (pack.selected) {
+                pack.selected = false;
+                pack.isHidden = false;
+                pack.name.includes(filterAvailable.current) ? newPackagesAvailable.push(pack) : null;
+                return false;
+            }
+
+            return true;
+        });
+
+        const updatedPackagesAvailable = [ ...newPackagesAvailable, ...packagesAvailable ];
+
+        updateState(updatedPackagesAvailable, updatedPackagesChosen);
+    };
+
+    const moveAllToChosen = () => {
+        const newPackagesChosen = packagesAvailable.reduce((result, pack) => {
+            pack.selected = false;
+            pack.isHidden = isHidden(filterChosen.current, pack);
+            return [ ...result, pack ];
+        }, []);
+
+        const updatedPackagesAvailable = [];
+        const updatedPackagesChosen = [ ...newPackagesChosen, ...packagesChosen ];
+
+        updateState(updatedPackagesAvailable, updatedPackagesChosen);
+    };
+
+    const moveAllToAvailable = () => {
+        const updatedPackagesChosen = packagesChosen.filter(pack => pack.isHidden);
+
+        const newPackagesAvailable = filterAvailable.current === undefined ? [] :
+            packagesChosen.reduce((result, pack) => {
+                pack.selected = false;
+                return (!pack.isHidden && pack.name.includes(filterAvailable.current) ? [ ...result, pack ] : [ ...result ]);
+            }, []);
```

Comment:
I would also write this using `.map`. It should make the return statement a bit clearer.